src/utils/debounce.ts

```diff
@@ -0,0 +1,24 @@
+export function debounce<T extends (...args: any[]) => void>(fn: T, wait: number = 0, leading?: boolean) {
+  let timer: number | ReturnType<typeof setTimeout>;
```

Comment:
Why is `ReturnType<typeof setTimeout>` needed here?

Reply:
I'm using takeover mode `Volar` instead of Vscode builtin Typescript
I'll check with Vscode Typescript


```ts
let timer: number; // only type number will cause typescript error

// Type 'Timeout' is not assignable to type 'number'.
timer = setTimeout(() => {
  fn.apply(this, args);
}, wait);
```

---

```ts
// ReturnType<typeof setTimeout> => number

let timer: number | ReturnType<typeof setTimeout>;

// no error
timer = setTimeout(() => {
  fn.apply(this, args);
}, wait);
```

Reply:
Ow, you're right :sweat:👍 
This is a client-side package